proto/cosmos/auth/query.proto

```diff
@@ -11,10 +12,14 @@ option go_package = "github.com/cosmos/cosmos-sdk/x/auth/types";
 // Query creates service with Account and Parameters as rpc
 service Query{
     // Account returns account details based on address
-    rpc Account (QueryAccountRequest) returns (QueryAccountResponse) {}
+    rpc Account (QueryAccountRequest) returns (QueryAccountResponse) {
+      option (google.api.http).get = "/cosmos/auth/account/{address}";
```

Comment:
Out of curiosity what is the reasoning to have /cosmos/?

Reply:
@marbar3778 there's no strong reason, we need some prefix to keep GRPC Gateway routes different from legacy routes. May be we can just replace it with version prefix

Reply:
I would prefer to not have the `/cosmos/` prefix. Some applications may not want this as well, and to change it would mean forking. could rest routes be versioned and then this would be as well. Current REST could be v0 or v1 and new REST routes v1 or v2? Sorry if I'm late to this discussion. If other people disagree then its fine

Reply:
For 0.40, the API legacy REST server will host 2 HTTP services on the same port:
- the legacy API
- this gRPC gateway

In 0.41, we'll remove the API legacy endpoints, and this port will serve 2 HTTP services:
- this gRPC gateway
- the rosetta api

The one requirement I have is that we don't change the gRPC gateway prefix between the 2 releases.

#### Proposal 1 (what Marko proposes)

```js
Legacy: `/v0/*`
GRPC: `/v1/*`
Rosetta: `/v1/rosetta/*` // Note: no app developer can have any SDK module called rosetta, but I guess that's fine
```

#### Proposal 2 (more flexbility imo)

```js
Legacy: `/*` // unchanged
GRPC: `/cosmos/v1/*` // also see: https://github.com/cosmos/cosmos-sdk/pull/6905#discussion_r464330757
Rosetta: `/rosetta/v1/*` // If ever rosetta api breaks, we can just update v1 to v2 here, without breaking anywhere else
```

Reply:
One alternative I have in mind is, replacing `cosmos` with `grpc` but that doesn't sound like a standard as well

Reply:
btw, whatever prefix we choose, we don't need to put the prefix in these annotations in any case: we could just mount the grpc-gateway on `/{chosen_prefix}/`

Reply:
The general logic here is I believe to follow the protobuf package name. I think that's the most sensible thing. Maybe we make a special exception for the `cosmos.*` packages, but I'd prefer consistency whatever we choose.

One option to consider is YAML configuration: https://grpc-ecosystem.github.io/grpc-gateway/docs/grpcapiconfiguration.html. I'm not sure if YAML will allow you to override annotations in the .proto files but that might be a way for apps to customize these routes. Also, we could decide to implement this whole thing with YAML files in the SDK rather than .proto files.

Reply:
> btw, whatever prefix we choose, we don't need to put the prefix in these annotations in any case: we could just mount the grpc-gateway on `/{chosen_prefix}/`

I don't think that's actually true. I think the generated swagger files will be for the wrong path then, although maybe there's a config parameter for that.